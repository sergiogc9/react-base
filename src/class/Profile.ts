export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
