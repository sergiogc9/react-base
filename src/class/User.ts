export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
