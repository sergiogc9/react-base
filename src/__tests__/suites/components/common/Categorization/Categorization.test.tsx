import { mount } from "enzyme";
import React from "react";
import Categorization from "@src/components/common/Categorization/Categorization";
import { TenantObject } from "@src/class/Tenant";

describe("Categorization", () => {
	let wrapper: any;
	let fetchTenantTagsMock: any;
	let onSetCategoryMock: any;
	let onRemoveTagMock: any;
	let onAddTagsMock: any;
	let onAddNotification: any;

	const tenant: TenantObject = {
		id: 'rd-girona-test',
		guid: '00034972-0000-0000-0000-000000000000',
		name: 'rd.girona.test',
		tier_properties: {
			name: 'custom',
			results: {
				online: true,
				social: true
			}
		},
		print_only: false,
		facebook_linked_ids: [
			'157193954975917'
		],
		settings: {
			categorization_mode: 'no_flc',
			currency: 'USD',
			display_influencers: true,
			facebook_url: 'https://www.facebook.com/conjunt.chapo',
			valuation_metric: 'miv'
		}
	};

	const availableTags = [
		{ tag: 'tag1', occurences: 2 },
		{ tag: 'tag2', occurences: 1 },
		{ tag: 'tag3', occurences: 8 }
	];

	beforeEach(() => {
		fetchTenantTagsMock = jest.fn();
		onSetCategoryMock = jest.fn();
		onRemoveTagMock = jest.fn();
		onAddTagsMock = jest.fn();
		onAddNotification = jest.fn();
		wrapper = mount(
			<Categorization
				tenant={tenant}
				tags={new Set('tag')}
				onAddTags={onAddTagsMock}
				onRemoveTag={onRemoveTagMock}
				onSetCategory={onSetCategoryMock}
				availableTags={availableTags}
				fetchTenantTags={fetchTenantTagsMock}
				onAddNotification={onAddNotification}
			/>
		);
	});

	it("Test component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot without tenant", () => {
		wrapper = mount(
			<Categorization
				tenant={null}
				tags={new Set('tag')}
				onAddTags={onAddTagsMock}
				onRemoveTag={onRemoveTagMock}
				onSetCategory={onSetCategoryMock}
				availableTags={availableTags}
				fetchTenantTags={fetchTenantTagsMock}
				onAddNotification={onAddNotification}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test select category", () => {
		expect(onSetCategoryMock).toHaveBeenCalledTimes(0);
		wrapper.find(".categorization-category-input").at(0).props().onAutocomplete(null, 0, [{ id: "010102" }]);
		expect(onSetCategoryMock).toHaveBeenCalledTimes(1);
	});

	it("Test select category input change", () => {
		expect(wrapper.state('categoryInput')).toBe('');
		wrapper.find(".categorization-category-input").at(0).props().onChange('test');
		wrapper.update();
		expect(wrapper.state('categoryInput')).toBe('test');
	});

	it("Test select category remove button", () => {
		wrapper.setState({ categoryInput: 'test' });
		wrapper.find("#categorizationCategoryRemoveButton").at(0).simulate('click');
		expect(wrapper.state('categoryInput')).toBe('');
	});

	it("Test set tag", () => {
		expect(onAddTagsMock).toHaveBeenCalledTimes(0);
		wrapper.find(".categorization-tags-input").at(0).props().onAutocomplete(null, 0, [{ value: "tag1" }]);
		expect(onAddTagsMock).toHaveBeenCalledTimes(1);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test remove tag", () => {
		wrapper.find(".categorization-tags-input").at(0).props().onAutocomplete(null, 0, [{ value: "tag2" }]);
		wrapper.update();
		expect(onRemoveTagMock).toHaveBeenCalledTimes(0);
		wrapper.find(".categorization-tags-selected .categorization-tags-selected-item-remove").at(0).simulate('click');
		expect(onRemoveTagMock).toHaveBeenCalledTimes(1);
	});

	it("Test on change tag input", () => {
		wrapper.find(".categorization-tags-input").at(0).props().onChange("test155");
		expect(wrapper.state().tagSuggested).toBe("test155");
	});

	it("Test tag too short notification", () => {
		wrapper.find(".categorization-tags-input").at(0).props().onAutocomplete(null, 0, [{ value: "aa" }]);
		expect(onAddNotification).toHaveBeenCalledTimes(1);
		expect(onAddNotification).toHaveBeenCalledWith({ t: 'results.categorize.error.minTagLength', level: 'warning' });
	});

	it("Test tag too large notification", () => {
		// eslint-disable-next-line max-len
		wrapper.find(".categorization-tags-input").at(0).props().onAutocomplete(null, 0, [{ value: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" }]);
		expect(onAddNotification).toHaveBeenCalledTimes(1);
		expect(onAddNotification).toHaveBeenCalledWith({ t: 'results.categorize.error.maxTagLength', level: 'warning' });
	});
});
