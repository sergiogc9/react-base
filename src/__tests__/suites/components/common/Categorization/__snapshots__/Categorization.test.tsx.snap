// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categorization Test component snapshot 1`] = `"<div class=\\"categorization\\"><div class=\\"categorization-category\\"><span class=\\"categorization-category-label\\">results.categorize.label.category</span><div id=\\"categoriesAutocompleter-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu-container--menu-below md-autocomplete-container categorization-category-input\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input md-autocomplete\\"><input autocomplete=\\"off\\" aria-autocomplete=\\"list\\" id=\\"categoriesAutocompleter\\" type=\\"text\\" class=\\"md-text-field md-text-field--margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div><button id=\\"categorizationCategoryRemoveButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">close</i></button></div><div class=\\"categorization-tags\\"><span class=\\"categorization-tags-label\\">results.categorize.label.tags</span><div id=\\"tagsAutocompleter-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu-container--menu-below md-autocomplete-container categorization-tags-input\\"><div class=\\"md-text-field-container md-inline-block md-text-field-container--input md-autocomplete\\"><label for=\\"tagsAutocompleter\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">results.categorize.tags.placeholder</label><input autocomplete=\\"off\\" aria-autocomplete=\\"list\\" id=\\"tagsAutocompleter\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div></div><div class=\\"categorization-tags-selected\\"><div class=\\"md-paper md-paper--1 categorization-tags-selected-item\\"><span>t</span><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block categorization-tags-selected-item-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div><div class=\\"md-paper md-paper--1 categorization-tags-selected-item\\"><span>a</span><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block categorization-tags-selected-item-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div><div class=\\"md-paper md-paper--1 categorization-tags-selected-item\\"><span>g</span><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block categorization-tags-selected-item-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div></div></div>"`;

exports[`Categorization Test component snapshot without tenant 1`] = `null`;

exports[`Categorization Test set tag 1`] = `"<div class=\\"categorization\\"><div class=\\"categorization-category\\"><span class=\\"categorization-category-label\\">results.categorize.label.category</span><div id=\\"categoriesAutocompleter-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu-container--menu-below md-autocomplete-container categorization-category-input\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input md-autocomplete\\"><input autocomplete=\\"off\\" aria-autocomplete=\\"list\\" id=\\"categoriesAutocompleter\\" type=\\"text\\" class=\\"md-text-field md-text-field--margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div><button id=\\"categorizationCategoryRemoveButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">close</i></button></div><div class=\\"categorization-tags\\"><span class=\\"categorization-tags-label\\">results.categorize.label.tags</span><div id=\\"tagsAutocompleter-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu-container--menu-below md-autocomplete-container categorization-tags-input\\"><div class=\\"md-text-field-container md-inline-block md-text-field-container--input md-autocomplete\\"><label for=\\"tagsAutocompleter\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">results.categorize.tags.placeholder</label><input autocomplete=\\"off\\" aria-autocomplete=\\"list\\" id=\\"tagsAutocompleter\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div></div><div class=\\"categorization-tags-selected\\"><div class=\\"md-paper md-paper--1 categorization-tags-selected-item\\"><span>t</span><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block categorization-tags-selected-item-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div><div class=\\"md-paper md-paper--1 categorization-tags-selected-item\\"><span>a</span><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block categorization-tags-selected-item-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div><div class=\\"md-paper md-paper--1 categorization-tags-selected-item\\"><span>g</span><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block categorization-tags-selected-item-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div></div></div>"`;
